### PR TITLE
Wrote unit tests for my REST API

### LINKED ISSUE #8: Write tests to ensure my API works correctly and my component(s) render correctly

### PR DESCRIPTION
**Pull request by allenhsieh**
This resolves #8 

1. Wrote 7 REST API unit tests that covers the routes currently available
2. Broke the server/index.js file into two files app.js and server.js to make testing the server logic possible without firing up the server itself. 
3. Removed mongoose connection from the model and put into my server code.

### CODE REVIEW

**Comment by allenhsieh** [comment]
@kogisu - I'm assigning you to be primary reviewer since I don't think i've gotten a review from you yet

File: server/app.test.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    test('it should respond with the index.html', (done) => {
+      request.get('/').then((response) => {
+        expect(response.text).toContain('id="app"');
+        done();
+      });
+    });
+
+    test('it should respond to a POST request with 404', (done) => {
+      request.post('/').then((response) => {
+        expect(response.statusCode).toBe(404);
```
Comment (kogisu) [inline server/app.test.js]: If this is meant as a rejection, this should be status code 401 as convention for 'unauthorized' use

File: server/app.test.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        expect(response.body.data.length).toBe(1);
+        done();
+      });
+    });
+
+    test('it should return 404 for an invalid endpoint', (done) => {
+      findByID.mockImplementation((id, cb) => {
+        cb('error', null);
+      });
+      request.get('/rooms/1').then((response) => {
```
Comment (kogisu) [inline server/app.test.js]: This endpoint is used on line 38.  Is this endpoint returning 404?

**Comment by kogisu** [comment]
Allen, looks pretty solid.  I liked that you connected the DB on the server and opened/ closed the server on the seed file.  Way to be modular!

**Comment by allenhsieh** [comment]
> Allen, looks pretty solid. I liked that you connected the DB on the server and opened/ closed the server on the seed file. Way to be modular!

Sounds good! feel free to squash and merge if you have time

File: server/app.test.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        expect(response.statusCode).toBe(404);
+        done();
+      });
+    });
+
+    test('response message for the invalid message should contain the id', (done) => {
+      request.get('/rooms/1').then((response) => {
+        expect(response.text).toBe('ID 1 does not exist in database');
```
Comment (allenhsieh) [inline server/app.test.js]: Actually hold up, i'm going to change this line based on a convo i had with philipp. My express route will return an error object, so i can render on page.

Comment (allenhsieh) [reply server/app.test.js]: @kogisu Ok pushed up, good for you to squash and merge for real now

Comment (kogisu) [reply server/app.test.js]: thanks! looks good

**Pull request merged**